Tools/Scripts/webkitpy/llvm_profile_utils.py

```diff
@@ -0,0 +1,148 @@
+                continue
+            if binary_path in llvm_profdata_binaries:
+                continue
+            llvm_profdata_binaries.append(binary_path)
+
+        logger.debug(f'Available {cls.EXECUTABLE_NAME} from {llvm_profdata_binaries}')
+
+        return llvm_profdata_binaries
+
+    @classmethod
+    def preference_ordered_paths(cls):
+        count = len(cls.detect_binaries())
+        for _ in range(count):
+            cls.PREFERRED_EXECUTABLE_INDEX = (cls.PREFERRED_EXECUTABLE_INDEX + 1) % count
+            yield cls.detect_binaries()[cls.PREFERRED_EXECUTABLE_INDEX]
+
+    @classmethod
+    def run(cls, command, *args, check=False, stdout=None, stderr=None, capture_output=False,
+            **kwargs) -> subprocess.CompletedProcess:
+        kwarg_capture_output = capture_output or (stdout is None and stderr is None)
+
+        completed_process = None
+        for binary_path in cls.preference_ordered_paths():
+            logger.debug(f'Running {shlex.join([binary_path, *command])}')
+            completed_process = subprocess.run([binary_path, *command], *args,
+                                               check=False, capture_output=kwarg_capture_output,
+                                               stdout=stdout, stderr=stderr, **kwargs)
+            if not completed_process.returncode:
+                break
+
+            logger.debug(f'Failed to {command} with binary {binary_path}\n'
+                         f'return_code: {completed_process.returncode}\n'
+                         f'stdout: {completed_process.stdout}\n'
+                         f'stderr: {completed_process.stderr}\n')
+
+        if check:
+            completed_process.check_returncode()
+
+        return completed_process
+
+
+class LLVMProfDataExecutable(ExecutablesFromEnvAndXcode):
+    EXECUTABLE_NAME = 'llvm-profdata'
+
+
+class LLVMProfileData:
+    @classmethod
+    def show(cls, profile_path):
+        list_functions_process = LLVMProfDataExecutable.run(['show', '--all-functions', '--value-cutoff=10',
+                                                             profile_path], stdout=subprocess.PIPE, text=True)
+
+        return subprocess.run(['c++filt', '-n'], input=list_functions_process.stdout,
+                              capture_output=True, text=True, check=True)
+
+    @classmethod
+    def merge(cls, output_file, unweighted_profiles=(), weighted_profiles=()):
+        command = ['merge', '--sparse', *unweighted_profiles]
+        for profile_path, weight in weighted_profiles:
+            lib_profile_path = profile_path
+            command.extend(['--weighted-input', f'{weight},{lib_profile_path}'])
+
+        command.extend(['--output', output_file])
+
+        return LLVMProfDataExecutable.run(command, capture_output=True, text=True)
+
+    @classmethod
+    def compress(cls, input_profile, output_file):
+        return subprocess.run(['/usr/bin/compression_tool', '-encode', '-i', input_profile, '-o', output_file,
```

Comment:
Are there reasons to be running `/usr/bin/compression_tool` and not `compression_tool`?

Reply:
Sure, just slightly surprised we're concerned about this here but not with any of the other commands we run.

Reply:
Ah, I missed `c++filt`, this should be fixed now.